$OpenBSD: Makefile $

Change Erlang version with the one set in the environment.

Index: Makefile
--- Makefile.orig
+++ Makefile
@@ -4,9 +4,9 @@
 MAN_PREFIX ?= $(SHARE_PREFIX)/man
 CANONICAL := v1.11/
 ELIXIRC := bin/elixirc --ignore-module-conflict $(ELIXIRC_OPTS)
-ERLC := erlc -I lib/elixir/include
-ERL_MAKE := if [ -n "$(ERLC_OPTS)" ]; then ERL_COMPILER_OPTIONS=$(ERLC_OPTS) erl -make; else erl -make; fi
-ERL := erl -I lib/elixir/include -noshell -pa lib/elixir/ebin
+ERLC := erlc${MODERL_VERSION} -I lib/elixir/include
+ERL_MAKE := if [ -n "$(ERLC_OPTS)" ]; then ERL_COMPILER_OPTIONS=$(ERLC_OPTS) erl${MODERL_VERSION} -make; else erl${MODERL_VERSION} -make; fi
+ERL := erl${MODERL_VERSION} -I lib/elixir/include -noshell -pa lib/elixir/ebin
 GENERATE_APP := $(CURDIR)/lib/elixir/generate_app.escript
 VERSION := $(strip $(shell cat VERSION))
 Q := @
@@ -27,7 +27,7 @@
 #==> Functions
 
 define CHECK_ERLANG_RELEASE
-	erl -noshell -eval '{V,_} = string:to_integer(erlang:system_info(otp_release)), io:fwrite("~s", [is_integer(V) and (V >= 21)])' -s erlang halt | grep -q '^true'; \
+	erl${MODERL_VERSION} -noshell -eval '{V,_} = string:to_integer(erlang:system_info(otp_release)), io:fwrite("~s", [is_integer(V) and (V >= 21)])' -s erlang halt | grep -q '^true'; \
 		if [ $$? != 0 ]; then \
 		  echo "At least Erlang/OTP 21.0 is required to build Elixir"; \
 		  exit 1; \
@@ -78,7 +78,7 @@
 	$(Q) cd lib/elixir && mkdir -p ebin && $(ERL_MAKE)
 
 $(PARSER): lib/elixir/src/elixir_parser.yrl
-	$(Q) erlc -o $@ +'{verbose,true}' +'{report,true}' $<
+	$(Q) erlc${MODERL_VERSION} -o $@ +'{verbose,true}' +'{report,true}' $<
 
 # Since Mix depends on EEx and EEx depends on Mix,
 # we first compile EEx without the .app file,
